src/util/resolveJsonOrRejectWithError.ts

```diff
@@ -0,0 +1,45 @@
+import defined from 'defined';
+
+export function createErrorPayload(status: number, messages: string, json: any) {
+  throw Object.assign(new Error(''), { status, json, messages }); // TODO: should be fixed in future
+}
+
+export const resolveLocation = (res: Response): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const location = res.headers.get('Location');
+    if (res.status === 201 && location) {
+      return resolve(location);
+    }
+    return reject(createErrorPayload(400, 'Location does not exist!', null));
```

Comment:
Hmm, kan være litt misvisende å returnere en `400` her fordi responsen ikke hadde location?

Reply:
Usikker på hvordan det burde håndteres. Hadde det vært bedre å returnere en tom streng? Isåfall vil kallene som er avhengige av location måtte sjekke om strengen er tom, noe som kanskje kunne vært greit?

Reply:
Tja, jeg liker egentlig bedre å rejecte med en error som vi allerede gjør, jeg liker bare ikke helt å lage en 400 siden det blir lett å tro at det kommer fra api'et om man ikke følger med. Hva med å bare sette status til 0 eller -1 eller noe som man ikke forventer av et apikall?

Reply:
Satt til -1 inntil videre.